rollup/types.go

```diff
@@ -368,33 +380,37 @@ func (c *CTCTxCreateEOA) Decode(b []byte) error {
 
 type CTCTxEIP155 struct {
 	Signature [65]byte
-	gasLimit  uint16 // uint16
-	gasPrice  uint8  // uint8
+	gasLimit  uint32 // uint24
+	gasPrice  uint32 // uint24
 	nonce     uint32 // uint24
 	target    common.Address
 	data      []byte
 }
 
 func (c *CTCTxEIP155) Len() (int, error) {
-	return 65 + 2 + 1 + 3 + 20 + len(c.data), nil
+	return 65 + 3 + 3 + 3 + 20 + len(c.data), nil
 }
 
-// TODO(mark): this serialization has fallen out of sync
-// with what is actually being submitted by the batch submitter
 func (c *CTCTxEIP155) Encode(b []byte) error {
 	length, _ := c.Len()
 	if len(b) < length {
 		return errors.New("CTCTxEIP155 encoding overflow")
 	}
 	copy(b[:65], c.Signature[:])
-	binary.BigEndian.PutUint16(b[65:65+2], c.gasLimit)
-	b[67] = c.gasPrice
-	hi := c.nonce & 0x00FF0000 >> 16
-	lo := c.nonce & 0x0000FFFF
+	hi := c.gasLimit & 0x00FF0000 >> 16
+	lo := c.gasLimit & 0x0000FFFF
+	b[65] = uint8(hi)
+	binary.BigEndian.PutUint16(b[66:68], uint16(lo))
+	hi = c.gasPrice & 0x00FF0000 >> 16
+	lo = c.gasPrice & 0x0000FFFF
 	b[68] = uint8(hi)
-	binary.BigEndian.PutUint16(b[69:69+2], uint16(lo))
-	copy(b[71:71+20], c.target.Bytes())
-	copy(b[91:], c.data)
+	binary.BigEndian.PutUint16(b[69:71], uint16(lo))
+	hi = c.nonce & 0x00FF0000 >> 16
+	lo = c.nonce & 0x0000FFFF
+	b[71] = uint8(hi)
+	binary.BigEndian.PutUint16(b[72:74], uint16(lo))
+	copy(b[74:94], c.target.Bytes())
+	copy(b[94:], c.data)
```

Comment:
😭 I'm sorry for 3 byte values

Reply:
Its ok, and a simple pattern to follow once it worked